cassandra-analytics-common/src/main/java/org/apache/cassandra/cdc/msg/CdcEventBuilder.java

```diff
@@ -0,0 +1,212 @@
+    {
+        // Skip updating TTL if it already has been set.
+        // For the same row, the upsert query can only set one TTL value.
+        if (timeToLive != null)
+        {
+            return;
+        }
+
+        setTimeToLive(new CdcEvent.TimeToLive(ttlInSec, expirationTimeInSec));
+    }
+
+    public void setTimeToLive(CdcEvent.TimeToLive timeToLive)
+    {
+        this.timeToLive = timeToLive;
+    }
+
+    public void setTombstonedCellsInComplex(Map<String, List<ByteBuffer>> tombstonedCellsInComplex)
+    {
+        this.tombstonedCellsInComplex = tombstonedCellsInComplex;
+    }
+
+    public void setRangeTombstoneList(List<RangeTombstone> rangeTombstoneList)
+    {
+        this.rangeTombstoneList = rangeTombstoneList;
+    }
+
+    public void setRangeTombstoneBuilder(RangeTombstoneBuilder<?> rangeTombstoneBuilder)
+    {
+        this.rangeTombstoneBuilder = rangeTombstoneBuilder;
+    }
+
+    public void setTrack(boolean track)
+    {
+        this.track = track;
+    }
+
+    public void setTrackingId(String trackingId)
+    {
+        this.trackingId = trackingId;
+        this.track = trackingId != null;
+    }
+
+    public void setKeyspace(String keyspace)
+    {
+        this.keyspace = keyspace;
+    }
+
+    public void setTable(String table)
+    {
+        this.table = table;
+    }
+
+    public void setKind(CdcEvent.Kind kind)
+    {
+        this.kind = kind;
+    }
+
+    public void setCassandraSource(CassandraSource cassandraSource)
+    {
+        this.cassandraSource = cassandraSource;
+    }
+
+
+    // adds the serialized cellpath to the tombstone
+    public void addCellTombstoneInComplex(String columnName, ByteBuffer key)
+    {
+        if (tombstonedCellsInComplex == null)
+        {
+            tombstonedCellsInComplex = new HashMap<>();
+        }
+        List<ByteBuffer> tombstones = tombstonedCellsInComplex.computeIfAbsent(columnName, k -> new ArrayList<>());
+        tombstones.add(key);
+    }
+
+    // Update the maxTimestamp if the input `timestamp` is larger.
+    protected void updateMaxTimestamp(long timestamp)
+    {
+        maxTimestampMicros = Math.max(maxTimestampMicros, timestamp);
+    }
+
+    public void validateRangeTombstoneMarkers()
+    {
+        if (rangeTombstoneList == null)
+        {
+            return;
+        }
+
+        Preconditions.checkState(!rangeTombstoneBuilder.hasIncompleteRange(),
+                                 "The last range tombstone is not closed");
+    }
+
+    public CdcEvent build()
+    {
+        validateRangeTombstoneMarkers();
+        return new CdcEvent(
+        partitionKeys,
+        clusteringKeys,
+        staticColumns,
+        valueColumns,
+        maxTimestampMicros,
+        timeToLive,
+        tombstonedCellsInComplex,
+        rangeTombstoneList,
+        track,
+        trackingId,
+        keyspace,
+        table,
+        kind,
+        cassandraSource
+        );
```

Comment:
nit: generally easier to read with passing the builder instance to the constructor, instead of having a long parameter list.